src/Compilers/Core/Portable/Collections/UnionCollection.cs

```diff
@@ -42,7 +42,24 @@ public static ICollection<T> Create(ICollection<T> coll1, ICollection<T> coll2)
                 return coll1;
             }
 
-            return new UnionCollection<T>(ImmutableArray.Create(coll1, coll2));
+            return new UnionCollection<T>(Flatten(ImmutableArray.Create(coll1, coll2)));
+        }
+
+        public static ICollection<T> Create(ImmutableArray<ICollection<T>> collections)
+        {
+            Debug.Assert(collections.All(c => c.IsReadOnly));
+
+            switch (collections.Length)
+            {
+                case 0:
+                    return SpecializedCollections.EmptyCollection<T>();
+
+                case 1:
+                    return collections[0];
+
+                default:
+                    return new UnionCollection<T>(Flatten(collections));
+            }
```

Comment:
nit: 

```c#
=> collections.Length switch
{
    0 => ...
}
```

Reply:
Not sure what the compiler style here is.

Reply:
they def use switch expressions.  but nbd.